src/Tuura/Concept/FSM/Translation.hs

```diff
@@ -110,16 +118,31 @@ handleArcs arcLists = result
             transCauses = cartesianProduct effectCauses
             result = map (\m -> (m, effect)) transCauses
 
-genFSM :: Show a => [a] -> [a] -> [a] -> [String] -> String -> String
-genFSM inputSigns outputSigns internalSigns arcStrs initialState =
-     printf tmpl (unwords ins) (unwords outs) (unwords ints) (unlines arcStrs) initialState
+genFSM :: Show a => [a] -> [a] -> [a] -> [String] -> String -> String -> String
+genFSM inputSigns outputSigns internalSigns arcStrs initialState reachReport =
+     printf tmpl (unwords ins) (unwords outs) (unwords ints) (unlines arcStrs) initialState reachReport
+    where
+      outs = map show outputSigns
+      ins = map show inputSigns
+      ints = map show internalSigns
+
+reachableInvariantStateError :: Show a => [a] -> String
+reachableInvariantStateError es = "Error:\n" ++
+                                  "The following state(s) are reachable but the invariant does not hold for them:\n" ++
+                                  unlines (errStates)
+    where
+      errStates = map (\e -> "s" ++ (show e)) es
```

Comment:
In such cases list comprehension usually works better: `errStates = [ "s" ++ show e | e <- es ]`.